internal/service/searchindex/resource_search_index.go

```diff
@@ -545,35 +543,35 @@ func validateSearchAnalyzersDiff(k, old, newStr string, d *schema.ResourceData)
 	return true
 }
 
-func unmarshalSearchIndexMappingFields(mappingString string) (map[string]any, diag.Diagnostics) {
-	if mappingString == "" {
-		return nil, nil
+func unmarshalSearchIndexMappingFields(str string) (map[string]any, diag.Diagnostics) {
```

Comment:
maybe an opportunity to unit test all these changed methods?

Reply:
answered below